### DISCUSSION TITLE
why do the placeholders not work 99% of the time?

### DISCUSSION THREAD

**Discussion by Fjox** [plan-player-analytics/Plan#2877] [Q&A]
i want to use plan analytics for leader boards and that stuff but the place holders dont work... papi is installed.-

**Comment by AuroraLS3** [comment]
Put the placeholders you are using to the Load_placeholders_on_join list in the Plan config

The placeholders can't be shown right away on server thread since waiting for database connection could crash the server

**Comment by Fjox** [comment]
i dont see that option in my config

**Reply by AuroraLS3** [reply]
It is under Plugins Placeholderapi in plan config

**Comment by Fjox** [comment]
didnt work

**Comment by Fjox** [comment]
![2023-04-24_19 00 25](https://user-images.githubusercontent.com/70401056/234065690-9165cffb-6569-4d07-8aa5-de5dfd42240b.png)

**Reply by AuroraLS3** [reply]
you might need to add {slow} or {slowest} to the line with the placeholder to make it update every 5/10 seconds

**Reply by AuroraLS3** [reply]
These seem to be running the placeholder replacement on server thread, and since Plan gets the data from database, it can take a while to get the value. 

Unfortunately that means the result might not be ready when the text loads.
The alternative is that the server crashes (waiting for database on server thread causes the server to stop ticking), so the placeholder either needs to be updated multiple times, or the replacement needs to happen off the server thread (Asynchronously)

**Reply by Fjox** [reply]
also will placeholders work on the proxy?

**Reply by AuroraLS3** [reply]
PlaceholderAPI does not support Bungeecord
